File: Frontend/src/types/plant.ts
export interface CareInstruction {
  wateringFrequency?: string;
  wateringTips?: string;
  sunlightNeeds?: string;
  sunlightTolerance?: string;
  soilType?: string;
  pruningSeason?: string;
  fertilizingFrequency?: string;
  propagationMethod?: string;
  repottingSeason?: string;
  advantages?: string[];
}

export interface ToxicityInfo {
  toxicToHumans?: boolean;
  humanToxicIf?: string;
  humanToxicParts?: string;
  humanToxicityDetail?: string;
  toxicToDogs?: boolean;
  dogToxicIf?: string;
  dogToxicParts?: string;
  dogToxicityDetail?: string;
  toxicToCats?: boolean;
  catToxicIf?: string;
  catToxicParts?: string;
  catToxicityDetail?: string;
}

export interface CommonProblem {
  title: string;
  description?: string;
  imageUrl?: string;
}

export interface PlantDTO {
  id: number;
  scientificName: string;
  commonName: string;
  description?: string;
  heightRange?: string;
  spreadRange?: string;
  leafType?: string;
  plantingSeason?: string;
  difficultyLevel?: string;
  resistanceLevel?: string;
  maintenanceLevel?: string;
  temperatureRange?: string;
  hardinessZone?: string;
  usages?: string;
  adaptationStrategies?: string;
  historyLegend?: string;
  nameHistory?: string;
  symbolism?: string;
  careInstruction?: CareInstruction;
  toxicityInfo?: ToxicityInfo;
  commonProblems?: CommonProblem[];
}

File: Backend/src/main/java/com/example/leafsense/Entity/Plant.java
package com.example.leafsense.Entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.util.List;

@Entity
@Table(name = "plants")
@Getter
@Setter
public class Plant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true, nullable = false)
    private String scientificName;

    private String commonName;

    @Column(length = 2000)
    private String description;

    private String heightRange;
    private String spreadRange;
    private String leafType;
    private String plantingSeason;

    private String difficultyLevel;
    private String resistanceLevel;
    private String maintenanceLevel;

    private String temperatureRange;
    private String hardinessZone;

    @Column(length = 2000) private String usages;
    @Column(length = 2000) private String adaptationStrategies;
    @Column(length = 2000) private String historyLegend;
    @Column(length = 1000) private String nameHistory;
    @Column(length = 500) private String symbolism;

    @OneToOne(mappedBy = "plant", cascade = CascadeType.ALL, orphanRemoval = true)
    private CareInstruction careInstruction;

    @OneToOne(mappedBy = "plant", cascade = CascadeType.ALL, orphanRemoval = true)
    private ToxicityInfo toxicityInfo;

    @OneToMany(mappedBy = "plant", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<CommonProblem> commonProblems;
}